### PR TITLE
chore(package): Updating packages to build accurately and include typings

### LINKED ISSUE #207: Typings are not properly published on npm
**Describe the bug**
This is not a bug in the repo itself, but the typings sub directories are not properly published on npm.

If you clone this repository and run `yarn build`, the typings seem to be created correctly.

However, installing 7.2.0 `ls typings/{browser,karma,server}` yields nothing.

This causes an error when trying to import, for example: `import { CustomWebpackBrowserBuilder } from '@angular-builders/custom-webpack';`.

**To Reproduce**
1. `npm install @angular-builders/custom-webpack'`
2. ts-node -e "import { CustomWebpackBrowserBuilder } from '@angular-builders/custom-webpack';"

This yields `error TS2305: Module '".../node_modules/@angular-builders/custom-webpack/typings"' has no exported member 'CustomWebpackBrowserBuilder'`.

**Expected behavior**
The `typings` for browser/karma/server should be published on npm.

**Builder:**
 - custom-webpack@7.2.0

### PR DESCRIPTION
**Pull request by ajcrites**
Using files in package.json is preferred over using .yarnignore / .npmignore. This allows us to more specifically select how we want to create the packages for publishing.

This fixes #207 

The results for `npm pack` for each package:

```
npm notice
npm notice 📦   @angular-builders/custom-webpack@7.2.0
npm notice === Tarball Contents ===
npm notice 1.2kB  package.json
npm notice 609B   builders.json
npm notice 9.3kB  README.md
npm notice 849B   src/browser/index.js
npm notice 13.8kB src/browser/schema.json
npm notice 134B   src/custom-webpack-builder-config.js
npm notice 798B   src/custom-webpack-builder.js
npm notice 126B   src/custom-webpack-schema.js
npm notice 372B   src/index.js
npm notice 845B   src/karma/index.js
npm notice 7.4kB  src/karma/schema.json
npm notice 874B   src/server/index.js
npm notice 8.2kB  src/server/schema.json
npm notice 1.2kB  src/webpack-config-merger.js
npm notice 867B   typings/browser/index.d.ts
npm notice 271B   typings/custom-webpack-builder-config.d.ts
npm notice 407B   typings/custom-webpack-builder.d.ts
npm notice 172B   typings/custom-webpack-schema.d.ts
npm notice 150B   typings/index.d.ts
npm notice 869B   typings/karma/index.d.ts
npm notice 915B   typings/server/index.d.ts
npm notice 309B   typings/webpack-config-merger.d.ts
npm notice === Tarball Details ===
npm notice name:          @angular-builders/custom-webpack
npm notice version:       7.2.0
npm notice filename:      angular-builders-custom-webpack-7.2.0.tgz
npm notice package size:  8.5 kB
npm notice unpacked size: 49.7 kB
npm notice shasum:        511e9c921ea8f313beefb67d165529d29513edc5
npm notice integrity:     sha512-E4EcYm1zI+lAq[...]f5sEpQKJNUfXg==
npm notice total files:   22
npm notice
angular-builders-custom-webpack-7.2.0.tgz
```

```
npm notice
npm notice 📦   @angular-builders/dev-server@7.2.1
npm notice === Tarball Contents ===
npm notice 1.2kB package.json
npm notice 265B  builders.json
npm notice 2.9kB README.md
npm notice 2.4kB src/generic/index.js
npm notice 5.2kB src/generic/schema.json
npm notice 299B  src/index.js
npm notice 667B  src/webpack-config-retriever.js
npm notice 985B  typings/generic/index.d.ts
npm notice 87B   typings/index.d.ts
npm notice 535B  typings/webpack-config-retriever.d.ts
npm notice === Tarball Details ===
npm notice name:          @angular-builders/dev-server
npm notice version:       7.2.1
npm notice filename:      angular-builders-dev-server-7.2.1.tgz
npm notice package size:  4.3 kB
npm notice unpacked size: 14.5 kB
npm notice shasum:        5e8cde1658c0b1eb064b3306e27641b6a5a3c7fc
npm notice integrity:     sha512-zK6Ibr6bNiPka[...]DR/1FLKku9QPg==
npm notice total files:   10
npm notice
angular-builders-dev-server-7.2.1.tgz
```

```
npm notice
npm notice 📦   @angular-builders/jest@7.2.1
npm notice === Tarball Contents ===
npm notice 1.0kB  package.json
npm notice 222B   builders.json
npm notice 6.3kB  README.md
npm notice 1.2kB  src/custom-config.resolver.js
npm notice 890B   src/default-config.resolver.js
npm notice 440B   src/jest-config/default-config.js
npm notice 1.0kB  src/jest-config/global-mocks.js
npm notice 147B   src/jest-config/mock-module.js
npm notice 169B   src/jest-config/setup.js
npm notice 1.2kB  src/jest-configuration-builder.js
npm notice 1.5kB  src/jest.builder.js
npm notice 798B   src/options-converter.js
npm notice 111B   src/schema.js
npm notice 18.9kB src/schema.json
npm notice 314B   typings/custom-config.resolver.d.ts
npm notice 296B   typings/default-config.resolver.d.ts
npm notice 267B   typings/jest-config/default-config.d.ts
npm notice 516B   typings/jest-configuration-builder.d.ts
npm notice 488B   typings/jest.builder.d.ts
npm notice 157B   typings/options-converter.d.ts
npm notice 114B   typings/schema.d.ts
npm notice === Tarball Details ===
npm notice name:          @angular-builders/jest
npm notice version:       7.2.1
npm notice filename:      angular-builders-jest-7.2.1.tgz
npm notice package size:  10.4 kB
npm notice unpacked size: 36.0 kB
npm notice shasum:        7b1c7055efed3fb7804d5ad25fe56379641b37fc
npm notice integrity:     sha512-ZuhOrivyzyCE7[...]wvqla3TKicG9w==
npm notice total files:   21
npm notice
angular-builders-jest-7.2.1.tgz
```

```
npm notice
npm notice 📦   @angular-builders/timestamp@7.2.0
npm notice === Tarball Contents ===
npm notice 494B  package.json
npm notice 247B  builders.json
npm notice 1.1kB LICENSE
npm notice 1.1kB README.md
npm notice 381B  src/schema.json
npm notice 1.1kB src/timestamp.builder.js
npm notice 440B  typings/timestamp.builder.d.ts
npm notice === Tarball Details ===
npm notice name:          @angular-builders/timestamp
npm notice version:       7.2.0
npm notice filename:      angular-builders-timestamp-7.2.0.tgz
npm notice package size:  2.4 kB
npm notice unpacked size: 4.8 kB
npm notice shasum:        d6f0d6a886b4541211dd38bee4ea93af35bffd43
npm notice integrity:     sha512-G+DTP9gXguK68[...]Pkwt1/b3u8G5A==
npm notice total files:   7
npm notice
angular-builders-timestamp-7.2.0.tgz
```

This should work with `npm publish`, but unfortunately it doesn't seem like `yarn publish` works properly possibly due to this: https://github.com/yarnpkg/yarn/issues/2822

If we want to stick with `yarn publish`, we'll have to update the `.yarnignore` files for packages that have typings in nested directories.

### CODE REVIEW

**Comment by meltedspark** [comment]
@ajcrites Thanks for investing your time, I really appreciate that!

**TL;DR;**
I propose to keep the `files` property as well as to update the `.yarnignore` until the issue is fixed in `yarn`. 

**The explanation:**
The build is failing because, well, the CI scripts are using `yarn pack` to form the package which is later installed in the E2E apps. And as you mentioned (kudos for the investigation), `yarn pack` sucks when it comes to using `files` property.

As far as I'm concerned the main implication of using `npm` for publishing is having two package managers installed instead of having just one of them. Here is why:  
I prefer using `yarn` as a package manager in general, but there were also specific reasons for using it here (there were some troubles with relative paths in _package.json_ scripts when it was called with `npm`). This means we still have to use `yarn` for maintaining the dependencies, while with this `publish` problem we also have to use `npm` for publishing. And this, in turn means:

1. `.travis.yml` has to be updated. Since E2E tests should use the exact same artifact that is published, CI environment will also have to install both package managers (to use `npm pack` when building the packages for E2E). Not a big deal, but still...
2. I (or any other publisher) will have to remember using `npm` for publishing and `yarn` for maintaining dependencies.
3. All the CI scripts will have to be updated (to use `npm pack` and not `yarn pack`)

These are all the implications that I can think of right now, but in general I would say that the obligation of having two package managers makes the whole project harder to maintain.  
How about keeping the `files` but also updating the `.yarnignore` until the issue with `files` is fixed in `yarn`?

**Comment by meltedspark** [comment]
The build fails with the same error (couldn't find _schema.json_), are you sure `yarn pack` works as expected? Have you verified that locally?

**Comment by ajcrites** [comment]
We might have a bit of a problem here: `yarn pack` and `npm pack` work very differently. With `files` the way it is this PR, `npm pack` works as expected. However, `yarn pack` seems broken (it seems to pack *everything*) regardless of whether `.yarnignore` is present.

Removing `package.json#files` and updating `.yarnignore` for `typings/**/*` works as expected for `yarn`. It seems like with the current state of yarn / npm, we can't have packaging work for both *or* I just don't know what I'm doing.

**Comment by meltedspark** [comment]
Looks like fixing _.yarnignore_ and omitting `files` from _package.json_ is the only viable solution for now. Would you mind updating the pull request to make the build pass so that I could merge it and publish a new version?

**Comment by ajcrites** [comment]
I'll close this and create a new PR since most of the history won't be needed.
